quickwit-search/src/search_stream/leaf.rs

```diff
@@ -140,75 +140,224 @@ async fn leaf_search_stream_single_split(
+        schema: &'a Schema,
+        index_config: &dyn IndexConfig,
+    ) -> crate::Result<SearchStreamRequestFields> {
+        // TODO make sure it's a fast field
+        let fast_field = schema
+            .get_field(&stream_request.fast_field)
+            .ok_or_else(|| {
+                SearchError::InvalidQuery(format!(
+                    "Field `{}` does not exist in schema",
+                    &stream_request.fast_field
+                ))
+            })?;
+
+        if !Self::is_fast_field(schema, &fast_field) {
+            return Err(SearchError::InvalidQuery(format!(
+                "Field `{}` is not a fast field",
+                &stream_request.fast_field
+            )));
         }
-        value_type => {
+
+        let timestamp_field = index_config.timestamp_field(schema);
+        let partition_by_fast_field = stream_request
+            .partition_by_field
+            .as_deref()
+            .and_then(|field_name| schema.get_field(field_name));
+
+        if partition_by_fast_field.is_some()
+            && !Self::is_fast_field(schema, &partition_by_fast_field.unwrap())
+        {
             return Err(SearchError::InvalidQuery(format!(
-                "Fast field type `{:?}` not supported",
-                value_type
+                "Field `{}` is not a fast field",
+                &stream_request.partition_by_field.as_deref().unwrap()
             )));
         }
+
+        Ok(SearchStreamRequestFields {
+            schema: schema.to_owned(),
+            fast_field,
+            partition_by_fast_field,
+            timestamp_field,
+        })
+    }
+
+    pub fn fast_field_types(&self) -> (Type, Option<Type>) {
+        (
+            self.schema
+                .get_field_entry(self.fast_field)
+                .field_type()
+                .value_type(),
+            self.partition_by_fast_field
+                .map(|field| self.schema.get_field_entry(field).field_type().value_type()),
+        )
+    }
+
+    pub fn fast_fields_for_request(&self) -> HashSet<String> {
+        let mut set = HashSet::new();
+        set.insert(self.fast_field_name().to_string());
+        if let Some(timestamp_field) = self.timestamp_field_name() {
+            set.insert(timestamp_field.to_string());
+        }
+        if let Some(partition_by_fast_field) = self.partition_by_fast_field_name() {
+            set.insert(partition_by_fast_field.to_string());
+        }
+        set
+    }
+
+    pub fn timestamp_field_name(&self) -> Option<&str> {
+        self.timestamp_field
+            .map(|field| self.schema.get_field_name(field))
+    }
+
+    pub fn fast_field_name(&self) -> &str {
+        self.schema.get_field_name(self.fast_field)
+    }
+
+    pub fn partition_by_fast_field_name(&self) -> Option<&str> {
+        self.partition_by_fast_field
+            .map(|field| self.schema.get_field_name(field))
+    }
+
+    fn is_fast_field(schema: &Schema, field: &Field) -> bool {
```

Comment:
nitpick: could be a function check_fast_field that we can call like `Self::check_fast_field()?`.